src/app/shared/shared.module.ts

```diff
@@ -4,20 +4,29 @@ import { CommonModule } from '@angular/common';
 import { UserComponent } from './user/user.component';
 import { UsersListComponent } from './users-list/users-list.component';
 import { UsersDetailsComponent } from './users-details/users-details.component';
+import { UpdateFormComponent } from './forms/update-form/update-form.component';
+import { CreateFormComponent } from './forms/create-form/create-form.component';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
 
 @NgModule({
   declarations: [
     UserComponent, 
     UsersListComponent, 
-    UsersDetailsComponent
+    UsersDetailsComponent, 
+    UpdateFormComponent,
+    CreateFormComponent
```

Comment:
The `form` used for creating / editing a user is same. So you can have a common component which contains this form. This component should be responsible for displaying the form-fields, doing the validations and emitting the form `submit` event.

The wrapper-components (Create and Update) which uses the above component will listen to this event and make the api call to the server. After success, re-direct the user back to page where user came from

Reply:
made a same for both of them